frontend/src/components/UserProfile.tsx

```diff
@@ -1,45 +1,52 @@
-import {
-  Box,
-  Button,
-  Flex,
-  ListItem,
-  Text,
-  UnorderedList,
-} from "@chakra-ui/react";
+import { Avatar, Button, Center, Flex, Text } from "@chakra-ui/react";
+import { useAsync } from "react-async-hook";
 import { Link } from "react-router-dom";
+import { getUserDetails } from "../services/songs";
 
 export default function UserProfile() {
+  const asyncUser = useAsync(async () => getUserDetails(), []);
+  const user = asyncUser.result && asyncUser.result.data;
+  const joinedDate =
+    user && user.date_joined
+      ? new Date(user?.date_joined)
+      : new Date(Date.now());
   return (
     <>
-      <Text
-        fontSize="3rem"
-        align="center"
-        fontFamily="Ubuntu Mono"
-        color="blue.600"
-        pt="2rem"
-        pb="2rem"
-      >
-        Your User Profile
-      </Text>
-      <Flex justifyContent="center">
-        <Box>
-          <Text mb="1rem" fontFamily="Ubuntu Mono" fontSize="1.5rem">
-            Upcoming personalizations:
-          </Text>
-          <UnorderedList mb="2rem" alignItems="center">
-            <ListItem>Toggle Night Mode</ListItem>
-            <ListItem>Toggle Chord Display</ListItem>
-            <ListItem>Set 1, 2, or 3 Columns</ListItem>
-            <ListItem>Your Requested Songs</ListItem>
-            <ListItem>Your Songbooks</ListItem>
-          </UnorderedList>
-          <Flex justifyContent="center">
+      {user ? (
```

Comment:
Meant to mention this while we were pairing but I forgot. We should probably use either a Skeleton or Spinner here. Spinner is easier, Skeleton probably gives a slightly better UX (but I don't remember what this part of the page looks like, so I don't know how easy it'd be to use here).

Reply:
It is done.